doc/specs/#5000 - Process Model 2.0/#5000 - Process Model 2.0.md

```diff
@@ -0,0 +1,1253 @@
+window process can attach that swapchain to the `SwapChainPanel` in the window.
+
+This is a little tricky, because WinRT does not natively expose a `HANDLE` type
+which would allow us to easily pass the `HANDLE` between these processes.
+Instead we'll need to manually cast the `HANDLE` to `uint64_t` at the winRT
+boundary, and cast it back to a `HANDLE` when we want to use it.
+
+We can't just have the content process duplicate the handle to the window
+process directly. If the content process is unelevated, and the window process
+is elevated (in a mixed elevation scenario), then the content process won't have
+permission to `DuplicateHandle` the `HANDLE` to the swapchain into the window's
+process. (more on mixed elevation below).
+
+Instead, the window will always need to be the one responsible for calling
+`DuplicateHandle`. Fortunately, the `DuplicateHandle` function does allow a
+caller to duplicate from another process into your own process.
+
+So when the content process needs to create a new swapchain, it'll raise an
+event that the window process can listen for to indicate that the swapchain
+changed. The window will then query for the content process's PID (which it will
+use to create a handle to the content process), and the window will query the
+current value of the content process's `HANDLE` to the swapchain. The window
+will then duplicate that `HANDLE` into it's own process space. Now that the
+window has a handle to the swapchain, it can use
+[`ISwapChainPanelNative2::SetSwapChainHandle`] to set the SwapChainPanel to use
+the same swapchain.
+
+This will allow the content process to draw to the swapchain, and the window
+process to render that same swapchain.
+
+#### Scenario: Tab Tear-off and Reattach
+
+Because all that's needed to uniquely identify an individual terminal instance
+is the GUID of the content process, it becomes fairly trivial to be able to
+"move" a terminal instance from one window to another.
+
+![drop-tab-on-existing-window](drop-tab-on-existing-window.png)
+
+When a drag/drop operation happens, the payload of the event will simply contain
+the structure of the tree of panes, and the GUIDs of the content processes that
+make up the leaf nodes of the tree. The receiving window process will then be
+able to use that tree to be able to re-create a similar pane structure in its
+window, and use the GUIDs to be able to connect to the content processes for the
+terminals in that tab.
+
+The terminal buffer never needs to move from one process to another - it always
+stays in just a single content process. The only thing that changes is the
+window process which is rendering the content process's swapchain.
+
+Similar to dragging a tab from one window to another window, we can also detect
+when the tab was dropped somewhere outside the bounds of a tab strip. When that
+happens, we'll create a new WT window, and use the data package from the
+drag-drop to initialize the structure of the window.

```

Comment:
It's not a big deal, but will the new window inherit the maximized/non-maximized state of the old window?

Reply:
Yes, it absolutely should. I'll make sure to call that out ☺️